daemon/daemon.go

```diff
@@ -1056,7 +1058,10 @@ func writeDistributionProgress(cancelFunc func(), outStream io.Writer, progressC
 
 	for prog := range progressChan {
 		if err := progressOutput.WriteProgress(prog); err != nil && !operationCancelled {
-			logrus.Errorf("error writing progress to client: %v", err)
+			// don't log broken pipe errors as this is the normal case when a client aborts
+			if !isBrokenPipe(err) {
+				logrus.Errorf("error writing progress to client: %v", err)
```

Comment:
Let's say you pull two different tags from different terminals, `image:a` and `image:b`. If you cancel `image:a`, `image:b` will continue, even if it shares some layers with `image:a`, but you're still cancelling the pull of `image:a`. That's why I think this is the appropriate place to log, so there's a record of why `image:a`'s pull was cancelled.

We could also log when individual layer downloads are cancelled (due to all pulls that need them being cancelled). This log message would go in `distribution/xfer/download.go` (see the comment `If an error was returned because the context was cancelled, we shouldn't retry`). But I don't think these log messages would be as useful, and there would be a lot of noise when you cancel a pull that's downloading several layers that no other pull is waiting for.